src/TSHelper.ts

```diff
@@ -74,7 +74,18 @@ export class TSHelper {
                 && (typeNode as ts.UnionOrIntersectionTypeNode).types.some(this.isArrayTypeNode));
     }
 
-    public static isArrayType(type: ts.Type, checker: ts.TypeChecker): boolean {
+    public static isArrayType(type: ts.Type, checker: ts.TypeChecker, dontCheckBases?: boolean): boolean {
```

Comment:
In what case would we **not** want to check base classes and interfaces?

Reply:
Maybe to avoid confusion the parameter could be renamed to `nonDerived` or we could create a new method `isPureArrayType(type,checker)` altogether. Not sure about the correct terminology, though.